### ISSUE TITLE
Creating A Buffer

### ISSUE THREAD

**Issue by jimswinder** [smeighan/xLights#3748]
**Is your feature request related to a problem? Please describe.**
Hard to create a Buffer using the drag method on a 180 degree pixel tree...at least I am having problems creating one to cover the exact pixels I need to include.

**Describe the solution you'd like**
Would it be possible to include an option to create a buffer using a group of nodes such as in creating a submodel?

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by merryoncherry** [comment]
First I'll admit that I'm not 100% sure what you're saying.  Is it the 3D nature of the tree that makes selection hard on the right side panel of the submodel dialog?  (Nodes wrapping around the side, or even the back if it is >360?)

As far as ideas you could try using xLights currently:
1. If it's a rectangle area you are after, try SubBuffer.
2. Flatten the tree out to less than 180 degrees while you do your work, then put it back when you are done.
3. Try the "Draw Model" button, this will let you draw on a flat grid instead.

**Comment by jimswinder** [comment]
I have a submodel that i am trying to buffer some of the pixels (per instructions from the Zoom guys). The submodel is 2 of my strands in the pixel tree...from the bottom pixels 14 thru 74 of a 100 ct strand. When trying to buffer it it keeps cutting out the middle pixels of one of the strands (per the House Preview).



Sent from my Verizon, Samsung Galaxy smartphone



-------- Original message --------
From: merryoncherry ***@***.***>
Date: 3/9/23 12:35 PM (GMT-08:00)
To: smeighan/xLights ***@***.***>
Cc: jimswinder ***@***.***>, Author ***@***.***>
Subject: Re: [smeighan/xLights] Creating A Buffer (Issue #3748)


First I'll admit that I'm not 100% sure what you're saying. Is it the 3D nature of the tree that makes selection hard on the right side panel of the submodel dialog? (Nodes wrapping around the side, or even the back if it is >360?)

As far as ideas you could try using xLights currently:

  1.  If it's a rectangle area you are after, try SubBuffer.
  2.  Flatten the tree out to less than 180 degrees while you do your work, then put it back when you are done.
  3.  Try the "Draw Model" button, this will let you draw on a flat grid instead.

**Comment by computergeek1507** [comment]
Are you using a "Subbuffer" submodel or selecting the individual nodes?

**Comment by jimswinder** [comment]
To be honest, I just learned how to use Buffers last night from Brian in the Zoom Room. He was trying to help me to have my strings "vibrate" on a bass guitar.
So was using the Buffer Tab under the Layer Settings and trying to manipulate the "Buffer Box" to capture the pixels I wanted to create a buffer.

I could send a screen capture of what I am seeing if that would help.
________________________________

**Comment by jimswinder** [comment]
So just saw where the "sub Buffer" is located...so no...was not using the Sub Buffer, was using the Node Range tab.
________________________________

**Comment by jimswinder** [comment]
Hey Keith....

Looks like you worked on the Superstring per our conversation in the Zoom Room a few weeks ago when you and Brian were presenting your new Guitar effect (which is AWESOME!)

I tried out the Superstring, but I still could not get it to do what I would want it to do...perhaps it never will...but looking to have 2, 3, or more colors show up in the string separately.
Using an LOR controller, so 4 different channels, one for each color, Red, Green, Blue and White. Did not see a way to set each of the 4 colors to a separate channel.
I tried all of the different Color Handling options, but it still just played one color instead of the two I had selected...such as had red and green selected, but just played Red.
Perhaps I am doing something wrong or maybe it is just not programmed the way I want it to be!!  lol

Thanks for any input.
